src/main/java/com/bobocode/orm/jdbc/JDBCRepository.java

```diff
@@ -0,0 +1,133 @@
+package com.bobocode.orm.jdbc;
+
+import static com.bobocode.orm.util.EntityUtils.extractFieldName;
+import static com.bobocode.orm.util.EntityUtils.extractId;
+import static com.bobocode.orm.util.EntityUtils.extractIdField;
+import static com.bobocode.orm.util.EntityUtils.getFieldsSortedByName;
+
+import com.bobocode.orm.annotation.Id;
+import com.bobocode.orm.context.PersistenceContext;
+import com.bobocode.orm.exception.OrmException;
+import com.bobocode.orm.util.EntityKey;
+import com.bobocode.orm.util.SqlUtils;
+import java.lang.reflect.Field;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.util.VisibleForTesting;
+
+@Slf4j
+@RequiredArgsConstructor
+public class JDBCRepository {
+
+  private final DataSource dataSource;
+  private final PersistenceContext context;
+
+  public <T> T findOneById(Class<T> entityType, Object id) {
+    var key = EntityKey.of(entityType, id);
+
+    if (context.contains(key)) {
+      T entity = context.getEntity(key);
+      log.info("Returning cached entity {}", entity);
+      return entity;
+    }
+
+    var idField = extractIdField(entityType);
+    var entities = findAllByField(entityType, idField, id);
+
+    checkOnlyOneEntityFound(entityType, id, entities);
+
+    return entities.get(0);
+  }
+
+  private void checkOnlyOneEntityFound(Class<?> entityType, Object id, List<?> entities) {
+    if (entities.isEmpty()) {
+      throw new IllegalStateException(
+          "No records found for '%s' entity with id=%s".formatted(entityType.getSimpleName(), id));
+    }
+    if (entities.size() > 1) {
+      throw new IllegalStateException(
+          "Found more than 1 record for entity %s with id=%s"
+              .formatted(entityType.getSimpleName(), id));
+    }
+  }
+
+  @VisibleForTesting
```

Comment:
Why do you need to test this method separately? Just to show and be sure that this functionality works in a correct way?

Reply:
Yep, the package-private modifier is needed to mock it for [this exceptional case](https://github.com/kebann/simple-orm/pull/1/files#diff-c6a263f1a499fb0ed8fc4bbdb07969e97a21e63f566ac815fdf201b129674f6a).
The test case was missing, so I've just added it.